### PR TITLE
17: Verify imported activation codes

### PR DESCRIPTION
**Pull request by sarahsporck**

### CODE REVIEW

File: frontend/lib/identification/activation_workflow/activation_code_parser.dart [hunk 1]
```diff
@@ -1,16 +1,12 @@
...
 import 'package:ehrenamtskarte/identification/qr_code_scanner/qr_code_processor.dart';
+import 'package:ehrenamtskarte/identification/verification_workflow/verification_qr_code_processor.dart';
+import 'package:ehrenamtskarte/identification/verification_workflow/verification_qr_content_parser.dart';
 import 'package:ehrenamtskarte/proto/card.pb.dart';
 
 class QRCodeInvalidTotpSecretException extends QrCodeParseException {
   QRCodeInvalidTotpSecretException() : super("invalid totp secret");
 }
 
-class QRCodeInvalidExpiryException extends QrCodeParseException {
```
Comment (sarahsporck) [inline frontend/lib/identification/activation_workflow/activation_code_parser.dart:10]: unused

**Review by f1sh1918 — APPROVED** [review]
Tested on android

File: frontend/lib/identification/activation_workflow/activation_code_scanner_page.dart [hunk 2]
```diff
@@ -27,8 +34,27 @@ class ActivationCodeScannerPage extends StatelessWidget {
...
+      final valid = await queryServerVerification(client, projectId, verifyQrCode);
+      if (!valid) {
+        await showError(
+          "Die zu prüfende Karte konnte vom Server nicht "
+          "verifiziert werden!",
+        );
+        return;
+      }
+
       provider.setCode(activationCode);
```
Comment (michael-markl) [inline frontend/lib/identification/activation_workflow/activation_code_scanner_page.dart:58]: Not sure, if you already handle this: If the user has no internet connection, a "nicer" message like "Bitte prüfen Sie Ihre Internetverbindung" should be shown. Also, optimally, before scanning the activation code, the user should be hinted that this action needs an internet connection.

Comment (sarahsporck) [reply frontend/lib/identification/activation_workflow/activation_code_scanner_page.dart:58]: This was basically copied from the verification workflow. I guess this rather checks whether the hash exists and such. So "Die prüfende Karte ist ungültig" would fit better.

**Review by maxammann — APPROVED** [review]

**Pull request merged**